### PR TITLE
feat: Implement point-along-line locations

### PR DESCRIPTION
**Pull request by UgnilJoZ**

### CODE REVIEW

File: openlr_dereferencer/decoding/point_locations.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+from ..maps.wgs84 import point_along_path
+from .line_decoding import dereference_path
+from . import LRDecodeError
+
+class PointAlongLine(NamedTuple):
+    """A dereferenced point along line location.
+
+    Contains the coordinates as well as the road on which it was located."""
+    line: Line
+    meters_into: float
```
Comment (ghost) [inline openlr_dereferencer/decoding/point_locations.py]: `positive_offset` or `meters_along` would be better

File: openlr_dereferencer/decoding/__init__.py [hunk 2]
```diff
@@ -25,7 +24,21 @@ def decode(reference: LineLocationRef, reader: MapReader, radius: float = SEARCH
...
 
     Returns:
-        The returned value will be a dereferenced openlr_dereferencer.LineLocation. It contains
-        a list of Line objects, which is trimmed by the offset values of the line location.
+        This function will return one or more map object, optionally wrapped into some class.
+        Here is an overview for what reference type will result in which return type:
+
+        reference                     | returns
+        ------------------------------|-------------------------------
+        openlr.GeoCoordinateLocation  | Node
```
Comment (ghost) [inline openlr_dereferencer/decoding/__init__.py:32]: maybe we can get the type names changed to `LineLocationReference` and `PointAlongLineLocationReference`

File: openlr_dereferencer/decoding/point_locations.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    side: SideOfRoad
+    orientation: Orientation
+
+    def location(self) -> Coordinates:
```
Comment (ghost) [inline openlr_dereferencer/decoding/point_locations.py]: `coordinates` might be a better name for the method

File: tests/test_decode.py [hunk 4]
```diff
@@ -180,25 +192,43 @@ def test_generate_candidates_1(self):
...
         self.assertAlmostEqual(path[1].lon, 13.414, delta=0.001)
         self.assertAlmostEqual(path[1].lat, 52.525, delta=0.001)
 
+    def test_decode_pointalongline(self):
+        "Test a valid point along line location"
+        # Get a list of 2 LRPs
+        reference = get_test_pointalongline()
+        pal: PointAlongLine = decode(reference, self.reader)
+        coords = pal.location()
+        self.assertAlmostEqual(coords.lon, 13.4153, places=4)
```
Comment (ghost) [inline tests/test_decode.py]: consistency in use of `places` and `delta` for assertions.

File: tests/test_decode.py [hunk 5]
```diff
@@ -180,25 +192,43 @@ def test_generate_candidates_1(self):
...
+        self.assertAlmostEqual(coords.lat, 52.5270, places=4)
+
+    def test_decode_pointalongline_raises(self):
+        "Test a valid point along line location with too high offset"
```
Comment (ghost) [inline tests/test_decode.py]: `invalid` rather than `valid`

File: openlr_dereferencer/maps/wgs84.py [hunk 6]
```diff
@@ -33,7 +33,7 @@ def project(point: Coordinates, dist: float, angle: float) -> Coordinates:
     # and `lat2` store the second point.
     return Coordinates(line['lon2'], line['lat2'])
 
-def point_along_line(path: Sequence[Coordinates], distance_meters: float) -> Coordinates:
+def point_along_path(path: Sequence[Coordinates], distance_meters: float) -> Coordinates:
```
Comment (ghost) [inline openlr_dereferencer/maps/wgs84.py]: if this is doing projection, maybe `project_along_path` might be a better name

**Review by ghost — APPROVED** [review]

File: openlr_dereferencer/decoding/__init__.py (hunk 2 above)
Comment (UgnilJoZ) [reply openlr_dereferencer/decoding/__init__.py:32]: - [ ] Let `python-openlr` names conform to whitepaper terminology

**Pull request merged**